.github/workflows/deploy-api-docs.yml

```diff
@@ -3,55 +3,82 @@ on:
-          HostedZoneName=vapor.codes,
-          AcmCertificateArn=${{ secrets.API_DOCS_CERTIFICATE_ARN }}
-      if: steps.cred.outcome == 'success'
-    - name: Deploy to S3
-      id: s3deploy
-      uses: jakejarvis/s3-sync-action@master
-      with:
-        args: --acl public-read --follow-symlinks
-      env:
-        AWS_S3_BUCKET: 'vapor-api-docs-site'
-        AWS_ACCESS_KEY_ID: ${{ secrets.API_DOCS_DEPLOYER_AWS_ACCESS_KEY_ID }}
-        AWS_SECRET_ACCESS_KEY: ${{ secrets.API_DOCS_DEPLOYER_AWS_SECRET_ACCESS_KEY }}
-        AWS_REGION: 'eu-west-2'
-        SOURCE_DIR: 'public'
-      if: steps.clouddeploy.outcome == 'success'
-    - name: Invalidate CloudFront
-      uses: awact/cloudfront-action@master
-      env:
-        SOURCE_PATH: '/*'
-        AWS_ACCESS_KEY_ID: ${{ secrets.API_DOCS_DEPLOYER_AWS_ACCESS_KEY_ID }}
-        AWS_SECRET_ACCESS_KEY: ${{ secrets.API_DOCS_DEPLOYER_AWS_SECRET_ACCESS_KEY }}
-        AWS_REGION: 'eu-west-2'
-        DISTRIBUTION_ID: ${{ secrets.VAPOR_API_DOCS_DISTRIBUTION_ID }}
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Build site
+        run: swift generate-api-docs.swift
+      - name: Install curl and awscliv2
+        run: | 
+          apt-get update && apt-get install -y curl
+          curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+          unzip awscliv2.zip
+          ./aws/install
+      - name: Regenerate all API docs if requested
+        if: inputs.regenerate_all
+        shell: bash
+        run: |
+          curl -sL \
+            https://raw.githubusercontent.com/vapor/api-docs/main/generate-package-api-docs.swift -o generate-package-api-docs.swift \
+            https://raw.githubusercontent.com/vapor/api-docs/main/theme-settings.json -o theme-settings.json            
+          for item in \
+            vapor:Vapor vapor:XCTVapor \
+            async-kit:AsyncKit routing-kit:RoutingKit console-kit:ConsoleKit \
+            websocket-kit:WebSocketKit multipart-kit:MultipartKit \
+            postgres-nio:PostgresNIO mysql-nio:MySQLNIO sqlite-nio:SQLiteNIO sql-kit:SQLKit \
+            postgres-kit:PostgresKit mysql-kit:MySQLKit sqlite-kit:SQLiteKit fluent-kit:FluentKit \
+            fluent-kit:FluentSQL fluent-kit:XCTFluent fluent:Fluent fluent-postgres-driver:FluentPostgresDriver \
+            fluent-mongo-driver:FluentMongoDriver fluent-mysql-driver:FluentMySQLDriver fluent-sqlite-driver:FluentSQLiteDriver \
+            redis:Redis queues-redis-driver:QueuesRedisDriver queues:Queues queues:XCTQueues \
+            leaf-kit:LeafKit leaf:Leaf \
+            jwt-kit:JWTKit jwt:JWT apns:APNS \
+          do
+            repo=${item%:*} module=${item#*:}
+            if [[ ! -d ${repo} ]]; then
+                git clone https://github.com/vapor/${repo}.git ${repo}
+            fi
+            cp theme-settings.json ${repo}
+            (cd ${repo} && swift ../generate-package-api-docs.swift ${repo} ${module})
```

Comment:
I'm not a fan of this, it's way too unreadable

Reply:
You're not wrong. I was also being quite lazy about it. There's also an argument to be made that we should do the "full" redeploy for almost any non-trivial push in this repo - even a change in the workflows can end up having a direct impact on what's actually rendered. And because of how DocC works, we can't "cheat" for things like "but I only changed `theme-settings.json`, can't we just update the copies in S3?" - most of the time that would probably work, but there are cases where it affects the "build" stage (especially the static hosting version). Might be safe for static assets like replacement icons though. Of course, I need to add the logic for the extra Inner Platform Effect layer around DocC - merging the "base" api-docs theme with any per-repo customizations - anyway; right now all it does is not overwrite a catalog's `theme-settings.json` if it has one (which doesn't do any good if the bits it copied from this repo are out of date).

The flip side of _that_, naturally, is it's time-consuming to re-build every darn repo, re-sync the entire S3 bucket, and re-invalidate the _entire_ CloudFront cache...

Reply:
Oh I agree on the need to be able to build all the docs. What I'm not a fan of is the bash-fu that no one else understands 😅

I'd much rather it be a swift script that we have a chance of maintaining